plugins/woocommerce/tests/e2e-pw/utils/api.js

```diff
@@ -13,6 +13,42 @@ if ( process.env.CONSUMER_KEY && process.env.CONSUMER_SECRET ) {
 	} );
 }
 
+/**
+ * Allow explicit construction of api client.
+ */
+const constructWith = ( consumerKey, consumerSecret ) => {
+	api = new wcApi( {
+		url: config.use.baseURL,
+		consumerKey,
+		consumerSecret,
+		version: 'wc/v3',
+	} );
+};
+
+const throwCustomError = (
+	error,
+	customMessage = 'Something went wrong. See details below.'
+) => {
+	throw new Error(
+		customMessage
+			.concat(
+				`\nResponse status: ${ error.response.status } ${ error.response.statusText }`
+			)
+			.concat(
+				`\nResponse headers:\n${ JSON.stringify(
+					error.response.headers,
+					null,
+					2
+				) }`
+			).concat( `\nResponse data:\n${ JSON.stringify(
+			error.response.data,
+			null,
+			2
+		) }
+` )
+	);
+};
```

Comment:
The `@woocommerce/woocommerce-rest-api` library uses `axios` when making API calls. Unfortunately, error messages from `axios` are quite vague. In order to get more detail from failed `api` requests, I added a function to catch the error and expose its relevant details -- very helpful when debugging in my experience with this PR.

Reply:
Agreed.  This is a really solid improvement, thank you!